packages/tasit-identity-contract/src/GnosisSafeUtils.js

```diff
@@ -0,0 +1,149 @@
+import Action from "tasit-action";
+const { Contract } = Action;
+import { ethers } from "ethers";
+
+const { utils: ethersUtils } = ethers;
+const { bigNumberify } = ethersUtils;
+
+// 100 gwei
+const GAS_PRICE = bigNumberify(`${1e11}`);
+
+export default class GnosisSafeUtils {
+  #contract;
+
+  constructor(contract) {
+    this.#contract = contract;
+  }
+
+  // Sign a hash using N signers
+  // Based on: https://github.com/gnosis/safe-contracts/blob/102e632d051650b7c4b0a822123f449beaf95aed/test/utils.js#L93
+  multiSign = (wallets, hash) => {
+    let signatures = "0x";
+
+    wallets.forEach(wallet => {
+      const signingKey = new ethers.utils.SigningKey(wallet.privateKey);
+      const sig = signingKey.signDigest(hash);
+      signatures += ethers.utils.joinSignature(sig).slice(2);
+    });
+
+    return signatures;
+  };
+
+  // Based on: https://github.com/gnosis/safe-contracts/blob/102e632d051650b7c4b0a822123f449beaf95aed/test/utilsPersonalSafe.js#L30
+  estimateFromSafeTxGas = async (to, value, data, operation) => {
+    const provider = this.#contract._getProvider();
+    const safeContractAddress = this.#contract.getAddress();
+
+    // Encode call to the contract's estimate gas function
+    const callToEstimate = this.encodeFunctionCall(
+      this.#contract,
+      "requiredTxGas",
+      [to, value, data, operation]
+    );
+
+    // The response will be return via error message
+    const estimateResponse = await provider.call({
+      to: safeContractAddress,
+      from: safeContractAddress,
+      data: callToEstimate,
+    });
+
+    // Extract estimation from revert error message
+    const estimatedGasHex = "0x" + estimateResponse.substring(138);
+    let estimatedGas = bigNumberify(estimatedGasHex);
+
+    // Extracted from Gnosis Safe test case (Why it's necessary?)
+    // Add 10k else we will fail in case of nested calls
```

Comment:
I think this answers the question of why it's necessary, right?

Reply:
Sorry, I'm making that question because of the test cases work even if that extra gas commented (last commit).

Reply:
Ah ok. We can discuss this here https://github.com/tasitlabs/TasitSDK/pull/217#discussion_r261259917

Resolving this